File: libs/lib_widget/src/main/java/com/example/william/my/core/widget/decoration/RItemDecorationDivider.java
package com.example.william.my.core.widget.decoration;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Rect;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.example.william.my.core.widget.R;

/**
 * RecyclerView分割线
 */
public class RItemDecorationDivider extends RecyclerView.ItemDecoration {

    private final Paint mPaint;

    private final float mDividerHeight;

    public RItemDecorationDivider(Context context) {
        this(context, R.color.colorDivider, 8);
    }

    public RItemDecorationDivider(Context context, int color) {
        this(context, color, 8);
    }

    public RItemDecorationDivider(Context context, float height) {
        this(context, R.color.colorDivider, height);
    }

    public RItemDecorationDivider(Context context, int color, float height) {
        this.mPaint = new Paint();
        this.mPaint.setAntiAlias(true);
        this.mPaint.setColor(ContextCompat.getColor(context, color));

        this.mDividerHeight = height;
    }

    /**
     * padding
     */
    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        super.getItemOffsets(outRect, view, parent, state);
        outRect.bottom = (int) mDividerHeight;
    }

    /**
     * background
     */
    @Override
    public void onDraw(@NonNull Canvas c, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        super.onDraw(c, parent, state);

        int left = parent.getPaddingLeft();
        int right = parent.getWidth() - parent.getPaddingRight();

        for (int i = 0; i < parent.getChildCount() - 1; i++) {
            View view = parent.getChildAt(i);
            c.drawRect(left, view.getBottom(), right, view.getBottom() + mDividerHeight, mPaint);
        }
    }

    /**
     * above
     */
    @Override
    public void onDrawOver(@NonNull Canvas c, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        super.onDrawOver(c, parent, state);
    }
}

File: basic/basic_lib/src/main/java/com/example/william/my/lib/recyclerview/itemdecoration/RItemDecorationDivider.kt
package com.sceneconsole.artefact.lib.base.recyclerview.itemdecoration

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Rect
import android.view.View
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView

/**
 * RecyclerView分割线
 */
class RItemDecorationDivider(
    context: Context,
    private val dividerHeight: Int,
    private val dividerColor: Int = 0xFFCCCC,
) : RItemDecoration() {

    private val mPaint: Paint = Paint()

    init {
        mPaint.isAntiAlias = true
        mPaint.color = ContextCompat.getColor(context, dividerColor)
    }

    /**
     * padding
     */
    override fun getItemOffsets(
        outRect: Rect,
        view: View,
        parent: RecyclerView,
        state: RecyclerView.State
    ) {
        super.getItemOffsets(outRect, view, parent, state)
        outRect.bottom = dividerHeight
    }

    /**
     * background
     */
    override fun onDraw(c: Canvas, parent: RecyclerView, state: RecyclerView.State) {
        super.onDraw(c, parent, state)
        val left = parent.paddingLeft
        val right = parent.width - parent.paddingRight
        for (i in 0 until parent.childCount - 1) {
            val view = parent.getChildAt(i)
            c.drawRect(
                left.toFloat(),
                view.bottom.toFloat(),
                right.toFloat(),
                (view.bottom + dividerHeight).toFloat(),
                mPaint
            )
        }
    }
}
